src/models/graph_tools.py

```diff
@@ -0,0 +1,77 @@
+"""
+This module contains the functions to build a networkx graph from polygon habitat data.
+"""
+
+
+from typing import Optional, List
+
+import networkx as nx
+import geopandas as gpd
+from shapely.strtree import STRtree
+from tqdm import tqdm
+
+
+def create_nx_graph(
+    gdf: gpd.GeoDataFrame, attributes: Optional[List[str]] = None
+) -> nx.Graph:
+    """
+    Creates networkx graph from GeoDataFrame. For each polygon in gdf a node (=vertex)
+    is created. Edges are created when the polygons of two nodes overlap or touch
+    borders. Largely taken from a notebook by @herbiebradley.
```

Comment:
I guess this file will become obsolete when the network `GeoGraph` object will be merged? 
Shall we push this file and add a #TODO to remember to remove it when we merge with the GeoGraph branch? (Or merge with the GeoGraph branch in the first place?)

```suggestion
    Create networkx graph from GeoDataFrame.   #TODO: Refactor to GeoGraph
    
    For each polygon in gdf a node (=vertex) is created. 
    Edges are created when the polygons of two nodes overlap or touch borders. 
    Largely taken from a notebook by @herbiebradley.
```

Reply:
Yep, I think adding the TODO is a good idea, that way we can have functioning demo version in the master now. It will be easy to replace it once the Graph class is ready.

Reply:
Agreed, we can just merge the two files when we add branch with  `GeoGraph` to master.